### PR TITLE
Merge propagation and api/propagation into api/propagators

### LINKED ISSUE #311: Suggestion: Rename non-api `propagation` package to `propagators`
This would resolve the naming conflict with `api/propagation`.

### PR DESCRIPTION
**Pull request by ferhatelmas**
Rename and merge propagation and api/propagation
to api/propagators.

Drop propagator suffix in general such that
TextFormatPropagator becomes TextFormat since
usage is propagators.TextFormat

fixes #311

### CODE REVIEW

**Comment by lizthegrey** [comment]
Please use more descriptive pull request names in the future explaining the change rationale, even if cut-pasted from the Issue you're fixing. Otherwise people reading the commit log may be confused.

File: api/propagators/b3_propagator.go [hunk 1]
```diff
@@ -35,7 +33,7 @@ const (
 	B3ParentSpanIDHeader = "X-B3-ParentSpanId"
 )
 
-// B3Propagator that facilitates core.SpanContext
+// B3 that facilitates core.SpanContext
```
Comment (lizthegrey) [inline api/propagators/b3_propagator.go]: Please update this documentation to make grammatical sense; e.g. `B3 propagator serializes core.SpanContext to/from B3 Headers.`

Comment (ferhatelmas) [reply api/propagators/b3_propagator.go]: updated.

**Comment by lizthegrey** [comment]
Update go.sum please via `make precommit`?

**Comment by ferhatelmas** [comment]
> Update go.sum please via make precommit?

Forgot pull before rebase, reverted now.

File: api/propagators/doc.go [hunk 2]
```diff
@@ -12,5 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Package propagation contains interface definition for Binary and TextFormat propagators.
-package propagation // import "go.opentelemetry.io/otel/api/propagation"
+// Package propagators contains interface definition for BinaryFormat and
+// TextFormat propagators and implementation of propagators for different
+// format and carriers.
```
Comment (rghetia) [inline api/propagators/doc.go]: s/carriers/suppliers/ - api has supplier.

Comment (ferhatelmas) [reply api/propagators/doc.go]: updated.

**Pull request merged**